src/lints/trait_associated_const_added.ron

```diff
@@ -0,0 +1,64 @@
+SemverQuery(
+    id: "trait_associated_const_added",
+    human_readable_name: "non-sealed trait added associated constant without default value",
+    description: "TODO",
+    required_update: Major,  // TODO
+    lint_level: Deny,  // TODO
+    reference_link: None,  // TODO
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Trait {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        sealed @filter(op: "!=", value: ["$true"])
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        associated_constant {
+                            associated_constant: name @output @tag
+                            public_api_eligible @filter(op: "=", value: ["$true"])
+                            default @filter(op: "=", value: ["$null"])
+                        }
+                    }
+                }
+            }
+            current {
```

Comment:
And this edge is looking at the new version.

You may want to flip the order of `baseline` and `current` in the query for this lint. It's easier to say "new has X and old doesn't" than to say "old doesn't have a thing that we'll look up later." (Actually the second one might be impossible in Trustfall, or at least very non-obvious.)